packages/core/src/App/Containers/OneTimeDepositModal/one-time-deposit-modal.tsx

```diff
@@ -29,15 +26,22 @@ const OneTimeDepositModal = observer(() => {
     } = ui;
     const liveChat = useLiveChat(false, loginid);
 
+    /** fetch account_status every 2 seconds since we can't subscribe for status updates */
     React.useEffect(() => {
-        setIsIframeLoading(true);
-    }, [iframe_url]);
+        const interval = setInterval(() => {
+            updateAccountStatus();
+        }, 2000);
+        return () => clearInterval(interval);
+    }, []);
 
     React.useEffect(() => {
-        if (is_deposit_locked) {
+        const has_client_deposited = account_status?.status?.some(
+            status => status === 'unwelcome' || status === 'withdrawal_locked' || status === 'cashier_locked'
+        );
```

Comment:
@farhan-nurzi-deriv Let's abstract this into a hook (`useHasDoneFirstDeposit` or something more meaningful) since it's already duplicated in this PR, Use the `useFetch` hook to fetch `get_account_status` and use the `refetchInterval` option instead of `setInterval` 🙏🏻

Reply:
@farzin-deriv addressed your suggestion 🙏